docs/development/extensions-core/spark.md

```diff
@@ -0,0 +1,275 @@
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~   http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+# Apache Spark Reader and Writer for Druid
+
+## Reader
+The reader reads Druid segments from deep storage into Spark. It locates the segments to read and determines their
+schema if not provided by querying the brokers for the relevant metadata but otherwise does not interact with a running
+Druid cluster.
+
+Sample Code:
+```scala
+val metadataProperties = Map[String, String](
+  "metadataDbType" -> "mysql",
+  "metadataConnectUri" -> "jdbc:mysql://druid.metadata.server:3306/druid",
+  "metadataUser" -> "druid",
+  "metadataPassword" -> "diurd"
+)
+
+sparkSession

```

Comment:
It would be good to also include an example that illustrates how to provide broker url for determining schema.

Reply:
Added 👍

Reply:
@JulianJaffePinterest , appreciate your effort to build something like this. Thank you !! .  I am trying to use this spark connector .  Just wanted to check if the - spark.md  file is merged to sprak-druid-connector branch?